Library/Homebrew/cmd/shellenv.sh

```diff
@@ -85,7 +85,7 @@ homebrew-shellenv() {
       then
         PATH_HELPER_ROOT="${PATH_HELPER_ROOT}" PATH="${HOMEBREW_PATH}" /usr/libexec/path_helper -s
       else
-        echo "export PATH=\"${HOMEBREW_PREFIX}/bin:${HOMEBREW_PREFIX}/sbin\${PATH+:\$PATH}\";"
+        echo "export PATH=\"${HOMEBREW_PREFIX}/bin:${HOMEBREW_PREFIX}/sbin:\$PATH\";"
```

Comment:
Yeah I think the case being hit here is not this line, it's the `/usr/libexec/path_helper` line above which is new.

The reason the issue has only been seen in GitHub Actions is because of https://github.com/actions/runner-images/blob/d1db39831d5ff8fe7737d54e947bc54a1b125c31/images/macos/scripts/build/configure-shell.sh#L17 which would take the path at the time of the image build. We normally recommend ` echo 'eval "$(/opt/homebrew/bin/brew shellenv)"' > ~/.bashrc` instead so other end users haven't noticed any users.

Reply:
Ah, yes. Instead of doing
```sh
/opt/homebrew/bin/brew shellenv >> ~/.bashrc
```
you should be doing
```sh
echo 'eval "$(/opt/homebrew/bin/brew shellenv)"' > ~/.bashrc
```
instead. (This is the same as what Bo said above; I just thought I'd make it more explicit.)

Reply:
> `echo 'eval "$(/opt/homebrew/bin/brew shellenv)"' > ~/.bashrc`

Looks valid - pretty sure it will solve the issue too, but why do we need the interpreted PATH construct at all if we can use containment with the original variable by default?

Reply:
> why do we need the interpreted PATH construct at all if we can use containment with the original variable by default?

Could you clarify what you mean by this? What is "the interpreted PATH construct"? How would we use "containment with the original variable"?

Reply:
> This also eplxains why it only affects macos-14 (path_helper isn't used on macos-13)

Let's say it affects `arm64`-based runners only. 😄 `macos-13` YAML label is Intel/amd64 based, so we don't use explicit `.bashrc` modifying there.

Reply:
For the specific line you modified: it was added for people who run with `bash -u`: https://github.com/Homebrew/brew/commit/59866d25ea39e4c5d25eb7cf78089d39752d88c3.

For the general behaviour of not using `$PATH` with `path_helper`. I agree it's not ideal. Just seems to be what Apple's `path_helper` tool outputs with no exceptions. The reason we're using `path_helper` is to support a new feature we added where people can add paths to `$(brew --prefix)/etc/paths`. I guess ideally we'd have something else that works on Linux too - we just prioritised the largest user share group for now.

Reply:
> Could you clarify what you mean by this? What is "the interpreted PATH construct"? How would we use "containment with the original variable"?

Yeah, sure. Now when you run `shellenv` you'll get something like this:

```
export HOMEBREW_CELLAR="/opt/homebrew/Cellar";
export HOMEBREW_REPOSITORY="/opt/homebrew";
PATH="/opt/homebrew/bin:/opt/homebrew/sbin:/Users/runner/.cargo/~~~~~~sr/appleinternal/bin"; export PATH;
[ -z "${MANPATH-}" ] || export MANPATH=":${MANPATH#:}";
export INFOPATH="/opt/homebrew/share/info:${INFOPATH:-}";
```

While more common way is to use next:
```
export HOMEBREW_CELLAR="/opt/homebrew/Cellar";
export HOMEBREW_REPOSITORY="/opt/homebrew";
export PATH="/opt/homebrew/bin:/opt/homebrew/sbin:$PATH"
```

But as @Bo98 described here we hit users of `bash -u` somehow this way.

Reply:
Anyway. You already sent a solution with `echo 'eval "$(/opt/homebrew/bin/brew shellenv)"' > ~/.bashrc`. Which is quite okay. I think we can drop this PR.

Reply:
> While more common way is to use next:
> 
> ```
> export HOMEBREW_CELLAR="/opt/homebrew/Cellar";
> export HOMEBREW_REPOSITORY="/opt/homebrew";
> export PATH="/opt/homebrew/bin:/opt/homebrew/sbin:$PATH"
> ```
> 
> But as @Bo98 described here we hit users of `bash -u` somehow this way.

Yes, this does hit users of `bash -u`. Moreover, doing
```
PATH="/foo/bar:/baz:/random/path"; export PATH;
```
is the output of `/usr/libexec/path_helper`, which we don't control. (Try running `/usr/libexec/path_helper -s` on a Mac to see.)